internal/driver/rest/api.go

```diff
@@ -0,0 +1,157 @@
+package rest
+
+import (
+	"encoding/json"
+	"github.com/isdzulqor/donation-hub/internal/core/service/project"
+	"github.com/isdzulqor/donation-hub/internal/core/service/user"
+	"github.com/isdzulqor/donation-hub/internal/driver/rest/req"
+	"github.com/jmoiron/sqlx"
+	"gopkg.in/validator.v2"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+type API struct {
+	DB             *sqlx.DB
+	UserService    user.Service
+	ProjectService project.Service
+}
+
+func (api *API) HandlePostUserRegister(w http.ResponseWriter, r *http.Request) {
+	log.Println("register")
+
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusNotFound)
+		_ = json.NewEncoder(w).Encode(NewNotFound())
+		return
+	}
```

Comment:
This HTTP method checking mechanism should not be done here. It should be easily defined on API list.

i.e
```go
r.Get("/topics", a.serveResolveTopics)
r.Route("/users", func(r chi.Router) {
r.Get("/", a.serveFindUsers)
r.Route("/{user_id}", func(r chi.Router) {
	r.Use(middleware.UserVerifier())
	r.Get("/", a.serveGetProfile)
	r.Put("/", a.serveUpdateUser)
```